util/printchplenv

```diff
@@ -39,260 +109,278 @@ def print_mode(mode='list', anonymize=False):
+        if chpl_tasks.get() == 'qthreads':
             link_args_3p.extend(chpl_3p_qthreads_configs.get_link_args())
 
-        print_var('  CHPL_RE2_UNIQ_CFG_PATH',
-                  chpl_3p_re2_configs.get_uniq_cfg_path(),
-                  mode, filters=('runtime',))
-        if regexp == 're2':
+        ENV_VALS['  CHPL_RE2_UNIQ_CFG_PATH'] = chpl_3p_re2_configs.get_uniq_cfg_path()
+        if chpl_regexp.get() == 're2':
             link_args_3p.extend(chpl_3p_re2_configs.get_link_args())
 
-        link_args_3p_dedup=[]
-        for arg in reversed(link_args_3p):
-            if arg not in link_args_3p_dedup:
-                link_args_3p_dedup.append(arg)
-        print_var('  CHPL_THIRD_PARTY_LINK_ARGS',
-                  ' '.join(reversed(link_args_3p_dedup)),
-                  mode, filters=('runtime',))
+        # Remove duplicates, keeping last occurrence and preserving order
+        # e.g. "-lhwloc -lqthread -lhwloc ..." -> "-lqthread -lhwloc ..."
+        seen = set()
+        link_args_3p_dedup = [arg for arg in reversed(link_args_3p)
+                              if not (arg in seen or seen.add(arg))]
+        ENV_VALS['  CHPL_THIRD_PARTY_LINK_ARGS'] = ' '.join(reversed(link_args_3p_dedup))
+
+
+"""Return non-empty string if var is set via environment or chplconfig"""
+def user_set(env):
+    env_set = overrides.get_environ(env, '')
+    config_set = overrides.get_chplconfig(env.strip(), '')
+    if env_set:
+        return ' *'
+    elif config_set:
+        return ' +'
+    return ''
+
+
+"""Filter variables that are not user set"""
+def filter_overrides(chpl_env):
+    return user_set(chpl_env.name)
+
+
+"""Filter variables irrelevant to configuration for --tidy flag"""
+def filter_tidy(chpl_env):
+    comm = ENV_VALS['CHPL_COMM']
+    if chpl_env.name == '  CHPL_COMM_SUBSTRATE':
+        return comm != 'none' and comm != 'ugni'
+    elif chpl_env.name == '  CHPL_GASNET_SEGMENT':
+        return comm == 'gasnet'
+    elif chpl_env.name == '  CHPL_NETWORK_ATOMICS':
+        return comm != 'none'
+    return True
+
+
+"""Filter CHPL_MAKE for --path (special case)"""
+def filter_path(chpl_env):
+    return chpl_env.name != 'CHPL_MAKE'
+
+
+"""Filter variables that are not selected in contents
+Requires a content argument via functools.partial
+"""
+def _filter_content(chpl_env, contents=None):
+    return chpl_env.content.intersection(contents)
+
+
+"""Return string to be printed for a given variable and print_format
+Requires a print_format argument
+"""
+def _print_var(env_var, value, print_format=None, shortname=None):
+    env = env_var.strip()
+    if print_format == 'pretty':
+        user_set_symbol = user_set(env)
+        return "{1}: {2}{0}\n".format(user_set_symbol, env_var, value)
+    elif print_format == 'simple':
+        return "{0}={1}\n".format(env, value)
+    elif print_format == 'make':
+        make_env_var = env_var.replace("CHPL_", "CHPL_MAKE_", 1).lstrip()
```

Comment:
why not
```
make_env_var = env.replace("CHPL_", "CHPL_MAKE_", 1)
```

i.e. you don't need lstrip if we start out with the stripped variable name?